src/Enum/exceptions/InvalidEnumValueException.php

```diff
@@ -15,21 +15,27 @@ class InvalidEnumValueException extends \Consistence\PhpException
 	/** @var mixed[] */
 	private $availableValues;
 
+	/** @var string */
+	private $className;
+
 	/**
 	 * @param mixed $value
 	 * @param mixed[] $availableValues
+	 * @param string $className
 	 * @param \Throwable|null $previous
 	 */
-	public function __construct($value, array $availableValues, \Throwable $previous = null)
+	public function __construct($value, array $availableValues, string $className, \Throwable $previous = null)
```

Comment:
I think it would be better to pass only $className. We would need to check that it is an enum class anyway, so calling `getAvailableValues()` on it statically should not be an issue as we have discussed in PMs.